test/playground/routes.js

```diff
@@ -121,26 +137,41 @@ exports.close = function (req, res) {
 
 exports.resize = function (req, res) {
     function resize (browser) {
-        return Promise.resolve()
-            .then(function () {
-                if (!browser.clientAreaSize)
-                    return Promise.resolve();
+        var requestedSize = getRequestedSize(req.body);
+
+        function resizeWindow () {
+            return browserNatives.resize(
+                browser.pageUrl,
+                browser.clientAreaSize.width,
+                browser.clientAreaSize.height,
+                requestedSize.width,
+                requestedSize.height
+            );
+        }
 
-                var args = [browser.pageUrl, browser.clientAreaSize.width, browser.clientAreaSize.height];
+        // NOTE: We must resize the window twice if it is maximized.
+        // https://github.com/DevExpress/testcafe-browser-natives/issues/71
+        return browserNatives
+            .isMaximized(browser.pageUrl)
+            .then(function (maximized) {
+                if (!maximized)
+                    return Promise.resolve();
 
-                if (req.body.paramsType === 'width-height')
-                    args = args.concat([Number(req.body.width), Number(req.body.height)]);
-                else {
-                    var deviceSize = browserNatives.getViewportSize(req.body.deviceName);
+                if (!browser.clientAreaSize)
+                    return Promise.reject(new Error(`Client area size is not found for the browser id-${browser.id}`));
```

Comment:
Sorry, i missed it - we can't use `` `string` `` with old versions of node.js
As i can see this package support them:
https://github.com/AndreyBelym/testcafe-browser-natives/blob/master/package.json#L14